pyFV3/stencils/delnflux.py

```diff
@@ -178,8 +179,8 @@ def diffusive_damp(
     damp: FloatFieldK,
 ):
     with computation(PARALLEL), interval(...):
-        fx = fx + 0.5 * damp * (mass[-1, 0, 0] + mass) * fx2
-        fy = fy + 0.5 * damp * (mass[0, -1, 0] + mass) * fy2
+        fx = fx + (0.5 * damp) * (mass[-1, 0, 0] + mass) * fx2
+        fy = fy + (0.5 * damp) * (mass[0, -1, 0] + mass) * fy2
```

Comment:
Will all phases of toolchain optimization and code generation strictly honor the added order of operations parentheses in these two statements?

Reply:
Yes. This strictly didn't need the parenthesis but in Fortran `0.5 * damp`  is pre-computed in front of the loop. For clarity I opted to show this works as a single statement. But the C rules here follow the Fortran rules and it's functionally equivalent

Reply:
A more complete answer would be that the optimization layer is _not_ allowed top break assign operation right hand side. So you will never have a pass that grabs `0.5*damp` figures it's a scalar that can be cached, and raise it up before the loop.
While this might look under-performing, generic compiler will do that pretty well if we code-gen flat and this way we allow for overcomputation on GPU, which is probably the fastest way to go.

Reply:
Appreciate the explanations as to how the toolchain will handle constructs like this when in the code generation phase (cuda, hip, c++, etc.)